### ISSUE TITLE
⌨ Screen reader not announcing lists correctly

### ISSUE THREAD

**Issue by yonikid15** [GCTC-NTGC/gc-digital-talent#16016]
### ⌨ Accessibility and Expected Behaviour

When navigating through the admin dashboard. A user experienced that the lists are reading the bullet separately from the link (e.g. "Bullet", tab, "Candidates", tab, "Bullet", tab "Community talent"). This is most likely happening on other places across the site.

It should be reading them as a whole (e.g. ""Unordered list, item 2, 'Actual Text'"

[Introduction to accessible lists and a screen reader demo](https://www.youtube.com/watch?v=Bc_hOdmVQc0&t=3s)

### 📋 Steps to Reproduce

1. Go to the admin dashboard
2. Navigate through the list items

### 🕵️ Details

Seems to be caused by tailwinds [list-disc property](https://github.com/GCTC-NTGC/gc-digital-talent/blob/14811-update-application-sidebar/apps/web/node_modules/@gc-digital-talent/ui/src/components/List/styles.ts#L7-L8) adding an additional item to the accessibility tree.

### What operating systems are you encountering the problem on?

iOS

### What browsers are you encountering the problem on?

Firefox

### What assistive technologies are you encountering the problem on?

VoiceOver

### 📸 Screenshot

<img width="2432" height="1252" alt="Image" src="https://github.com/user-attachments/assets/4f8f3614-f674-454c-a762-52c15bf90a27" />

<img width="1404" height="1241" alt="Image" src="https://github.com/user-attachments/assets/4c6c0eed-94d3-4d85-adbc-df265b6f3d71" />

### ✅ Acceptance Criteria

- [ ] Correctly announces list items on screen readers
